sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -1138,3 +1145,81 @@ export class MessageReceiver extends LinkEntity {
     return result;
   }
 }
+
+/**
+ * Wraps the receiver with some higher level operations for managing state
+ * like credits, draining, etc...
+ *
+ * @internal
+ * @ignore
+ */
+export class ReceiverHelper {
+  private _stopReceivingMessages: boolean = false;
+
+  constructor(private _getCurrentReceiver: () => Receiver | undefined) {}
+
+  /**
+   * Adds credits to the receiver, respecting any state that
+   * indicates the receiver is closed or should not continue
+   * to receive more messages.
+   *
+   * @param credits Number of credits to add.
+   */
```

Comment:
Nit: since you're returning a boolean, can you describe what it represents in the TSDoc?

Reply:
Done.